tests/phpt/expanders/nested_ast_expander.phpt

```diff
@@ -0,0 +1,20 @@
+--TEST--
+Provides an AST to a custom expander
+--FILE--
+<?php
+
+$(macro) {
+    $(\Yay\tests\fixtures\parsers\my_nested_ast_parser())
+} >> {
+    $$(\Yay\tests\fixtures\expanders\my_nested_ast_expander($(outer)))
```

Comment:
I think we need two combined nested expanders here in order to prove the composition is working:
```diff
>> {
-  $$(\Yay\tests\fixtures\expanders\my_nested_ast_expander($(outer)))+
+  $$(
+       \Yay\tests\fixtures\expanders\nyancat_ast_expander(
+           \Yay\tests\fixtures\expanders\uppercase_ast_expander($(outer))
+       )
+   )
}
```